packages/web3/src/contract/ralph.ts

```diff
@@ -385,6 +385,28 @@ export function primitiveToByteVec(value: Val, type: string): Uint8Array {
   }
 }
 
+export function typeLength(typ: string, structs: Struct[]): number {
+  if (isPrimitiveType(typ)) {
+    return 1
+  }
+
+  if (typ.startsWith('[')) {
+    const [baseType, size] = decodeArrayType(typ)
+    return size * typeLength(baseType, structs)
+  }
+
+  const struct = structs.find((s) => s.name === typ)
+  if (struct !== undefined) {
+    return struct.fieldTypes.reduce((acc, fieldType) => acc + typeLength(fieldType, structs), 0)
+  }
+
+  throw Error(`Invalid type ${typ}`)
```

Comment:
The compiler will return the contract name instead of `ByteVec` for the contract type, so we should return 1 here.